webapp/src/main/java/io/github/microcks/web/SoapController.java

```diff
@@ -200,6 +223,10 @@ public ResponseEntity<?> execute(
          if (!responses.isEmpty()) {
             int idx = DispatchStyles.RANDOM.equals(dispatcher) ? RandomUtils.nextInt(0, responses.size()) : 0;
             response = responses.get(idx);
+         } else {
+            return new ResponseEntity<String>(
+               String.format("The response %s does not exist!", dispatchContext.dispatchCriteria()),
+               HttpStatus.BAD_REQUEST);
```

Comment:
This is matching with my opinion in https://github.com/microcks/microcks/issues/819 😉 
We have to conclude as a community on this point, though.